crates/networking/p2p/utils.rs

```diff
@@ -82,6 +110,70 @@ pub fn dump_to_file(path: &Path, contents: Vec<u8>) -> Result<(), DumpError> {
         })
 }
 
+pub fn dump_accounts_to_file(
+    path: &Path,
+    accounts: Vec<(H256, AccountState)>,
+) -> Result<(), DumpError> {
+    cfg_if::cfg_if! {
+        if #[cfg(feature = "rocksdb")] {
+            dump_to_rocks_db(
+                path,
+                accounts
+                    .into_iter()
+                    .map(|(hash, state)| (hash.0.to_vec(), state.encode_to_vec())
+                ).collect::<Vec<_>>()
+            )
+                .inspect_err(|err| error!("Rocksdb writing stt error {err:?}"))
+                .map_err(|_| DumpError {
+                    path: path.to_path_buf(),
+                    contents: Vec::new(),
+                    error: std::io::ErrorKind::Other,
+                })
+        } else {
+            dump_to_file(path, accounts.encode_to_vec())
+        }
+    }
+}
+
+/// Struct representing the storage slots of certain accounts that share the same storage root
+pub struct AccountsWithStorage {
+    /// Accounts with the same storage root
+    pub accounts: Vec<H256>,
+    /// All slots in the trie from the accounts
+    pub storages: Vec<(H256, U256)>,
+}
+
+pub fn dump_storages_to_file(
+    path: &Path,
+    storages: Vec<AccountsWithStorage>,
+) -> Result<(), DumpError> {
+    cfg_if::cfg_if! {
```

Comment:
Please don't contribute to `cfg_if` proliferation. Just use `#[cfg]` at the blocks, or `if cfg!()` when possible to ensure the compiler checks all possibilities.